boa_parser/src/lexer/token.rs

```diff
@@ -99,7 +99,7 @@ pub enum TokenKind {
     EOF,
 
     /// An identifier.
-    Identifier(Sym),
+    Identifier((Sym, bool)),
```

Comment:
Would be good to have a `IsLegacyEscaped` enum instead of `bool`, because there are a lot of places where it is not obvious what the `bool` represents exactly.

Reply:
Good call, I added explicit types instead of the `bool`s.